src/rep/mesh.jl

```diff
@@ -0,0 +1,970 @@
+- `valid`                   -
+- `offset`                  - Offset indicating number to be added to migrate to 0-indexed system.
+
+- `_verts_len`              - Number of vertices in each mesh of TriMesh.
+- `_verts_list`             - Vertices in list format.
+- `_verts_packed`           - Vertices in packed format.
+- `_verts_padded`           - Vertices in padded format.
+
+- `_faces_len`              - Number of faces in each mesh of TriMesh.
+- `_faces_list`             - Vertices in list format.
+- `_faces_packed`           - Vertices in packed format.
+- `_faces_padded`           - Vertices in padded format.
+
+- `_edges_packed`           - Edges in packed format (according to packed vertices).
+- `_faces_to_edges_packed`  - Faces formed by edges in packed format (according to packed edges).
+- `_laplacian_packed`       - Laplacian sparce matrix in packed format.
+- `_edges_to_key`           - Dict mapping edges tuple to unique key.
+
+### Available Contructor:
+
+- `TriMesh(verts_list, faces_list; offset::Number = -1)`
+- `TriMesh(m::Vector{<:GeometryBasics.Mesh})`
+- `TriMesh(m::GeometryBasics.Mesh)`
+- `TriMesh(m::TriMesh)`
+
+"""
+mutable struct TriMesh{T<:AbstractFloat,R<:Integer} <: AbstractMesh{T,R}
+    N::Int64
+    V::Int64
+    F::Int64
+    equalised::Bool
+    valid::BitArray{1}
+    offset::Int8
+    _verts_len::AbstractArray{Int,1}
+    _faces_len::AbstractArray{Int,1}
+
+    _verts_packed::Union{Nothing,AbstractArray{T,2}}
+    _verts_padded::Union{Nothing,AbstractArray{T,3}}
+    _verts_list::Union{Nothing,Vector{<:AbstractArray{T,2}}}
+
+    _faces_packed::Union{Nothing,Array{R,2}}
+    _faces_padded::Union{Nothing,Array{R,3}}
+    _faces_list::Union{Nothing,Vector{<:AbstractArray{R,2}}}
+
+    _edges_packed::Union{Nothing,Array{R,2}}
+    _faces_to_edges_packed::Union{Nothing,Array{R,2}}
+    _laplacian_packed::Union{Nothing,AbstractSparseArray{T,R,2}}
+
+    _edges_to_key::Union{Nothing,Dict{Tuple{R,R},R}}
+end
+
+TriMesh(
+    verts::AbstractArray{<:AbstractArray{T,2},1},
+    faces::AbstractArray{<:AbstractArray{R,2},1};
+    offset::Number = -1,
+) where {T,R} = TriMesh(Float32.(verts), UInt32.(faces); offset = offset)
+
+function TriMesh(
+    verts::AbstractArray{<:AbstractArray{T,2},1},
+    faces::AbstractArray{<:AbstractArray{R,2},1};
+    offset::Number = -1,
+) where {T<:AbstractFloat,R<:Integer}
+
+    length(verts) == length(faces) ||
+        error("batch size of verts and faces should match, $(length(verts)) != $(length(faces))")
+    _verts_len = size.(verts, 1)
+    _faces_len = size.(faces, 1)
+    N = length(verts)
+    V = maximum(_verts_len)
+    F = maximum(_faces_len)
+    equalised = all(_verts_len .== V) && all(_faces_len .== F)
+    valid = _faces_len .> 0
+    offset = Int8(offset)
+
+    _verts_list = verts
+    _faces_list = faces
+
+    if verts[1] isa CuArray
+        _verts_len = gpu(_verts_len)
+        _faces_len = gpu(_faces_len)
+    end
```

Comment:
@avik-pal is there any better way to change to gpu

Reply:
No this is a bad idea. Type inference might fail in this case. The better way would be to store the type information directly in the `TriMesh` type and dispatch based on that.

Reply:
I was trying to add array type information in TriMesh type, but I was not able to extract array type name, like only (CuArray or Array, and not Array{Float32,2}). So is there any convenient function or workaround for this?

Reply:
I was suggesting something like 
```
TriMesh{T, R, S<:AbstractArray}
    ...
    verts_list::Union{Nothing, S}
    ....
```

Reply:
And while creating an instance use put the type information manually like `TriMesh{eltype(..), eltype(...), Array{....}}(....)`

Reply:
@nirmal-suthar Did you look into this?

Reply:
Yes, I am still working on this locally. Currently some tests are breaking, but will finish it soon.

Reply:
Can you suggest me a way to trim Array{Float32,2} -> Array. The former type is difficult to handle with different eltype and dims.